File: app/src/main/scala/tripaint/grid/ImageGridChange.scala
package tripaint.grid

import tripaint.coords.GridCoords
import tripaint.model.Change

class ImageGridChange(val changes: Map[GridCoords, ImageChange]) extends Change {
  override def undo(): Unit = for (_, change) <- changes do change.undo()

  override def redo(): Unit = for (_, change) <- changes do change.redo()
}

File: core/src/main/kotlin/tripaint/grid/ImageGridChange.kt
package tripaint.grid

import tripaint.Change
import tripaint.coords.GridCoords

class ImageGridChange(val changes: Map<GridCoords, ImageChange>) : Change {
    override fun undo() {
        for ((_, change) in changes) {
            change.undo()
        }
    }

    override fun redo() {
        for ((_, change) in changes) {
            change.redo()
        }
    }
}